app/common/lib/ledgerUtil.js

```diff
@@ -262,36 +262,58 @@ const getMediaData = (xhr, type) => {
   const parsedUrl = urlParse(xhr)
   const query = parsedUrl && parsedUrl.query
 
-  if (!parsedUrl || !query) {
+  if (!parsedUrl) {
     return null
   }
 
   switch (type) {
     case ledgerMediaProviders.YOUTUBE:
       {
-        result = queryString.parse(parsedUrl.query)
+        if (!query) {
+          return null
+        }
+
+        result = queryString.parse(query)
         break
       }
     case ledgerMediaProviders.TWITCH:
       {
-        result = queryString.parse(parsedUrl.query)
-        if (!result.data) {
+        const data = details.getIn(['uploadData', 0, 'bytes'])
```

Comment:
There's ALMOST enough logic here that we might consider making `youTubeUtils` / `twitchUtils` / etc and having a single method exposed (ex: `getTwitchData`).  This could allow for documenting what is being parsed in a clear way

Reply:
yeah we can refactor this in the next version, so that is easier for the further implementations